library/psa_crypto.c

```diff
@@ -1786,6 +1786,7 @@ static psa_status_t psa_start_key_creation(
      * definition. */
 
     slot->attr = attributes->core;
+    slot->status = PSA_SLOT_OCCUPIED;
```

Comment:
Same here, the slot shouldn't be marked as occupied when the function returns with an error code.